crates/polars-utils/src/not_nan.rs

```diff
@@ -0,0 +1,84 @@
+use std::hash::{Hash, Hasher};
+
+use num_traits::Float;
+use polars_error::{PolarsError, PolarsResult, polars_bail};
+
+/// A wrapper for floating point types to conform to `Eq` and `Hash`.
+///
+/// The implementation is inspired by the `ordered-float` crate.
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize), serde(transparent))]
+#[cfg_attr(
+    feature = "dsl-schema",
+    derive(schemars::JsonSchema),
+    schemars(transparent)
+)]
+pub struct NotNan<T>(T);
+
+/* ----------------------------------------- CONVERSION ---------------------------------------- */
+
+impl TryFrom<f32> for NotNan<f32> {
+    type Error = PolarsError;
+    fn try_from(value: f32) -> PolarsResult<Self> {
+        if value.is_nan() {
+            polars_bail!(ComputeError: "float is required to be not NaN")
+        }
+        Ok(NotNan(value))
+    }
+}
+impl From<NotNan<f32>> for f32 {
+    fn from(value: NotNan<f32>) -> Self {
+        value.0 + value.0
+    }
+}
+
+impl TryFrom<f64> for NotNan<f64> {
+    type Error = PolarsError;
+    fn try_from(value: f64) -> PolarsResult<Self> {
+        if value.is_nan() {
+            polars_bail!(ComputeError: "float is required to be not NaN")
+        }
+        Ok(NotNan(value))
+    }
+}
+
+impl From<NotNan<f64>> for f64 {
+    fn from(value: NotNan<f64>) -> Self {
+        value.0
+    }
+}
+
+/* ---------------------------------------- TRAIT IMPLS ---------------------------------------- */
+
+impl<F> Eq for NotNan<F> where F: Float {}
+
+impl Hash for NotNan<f32> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        // This hack is taken from the `ordered-float` crate:
```

Comment:
Actually, no, I added this to `ordered-float`, implementing it in Polars first. So it's the other way around.